src/server.rs

```diff
@@ -26,16 +31,24 @@ impl Server {
         self.set_logging_verbosity()
             .context("set logging verbosity")?;
 
-        let addr = "[::1]:50051".parse().context("parse server address")?;
         let rt = MyRuntime::default();
         let img = MyImage::default();
 
-        info!("Runtime server listening on {}", addr);
+        let sock_path = Path::new(self.config.sock_path());
+        if sock_path.exists() {
+            fs::remove_file(sock_path).await?;
+        } else {
+            fs::create_dir_all(sock_path.parent().unwrap()).await?;
```

Comment:
No need to unwrap here when using the `Context`:
```suggestion
            fs::create_dir_all(sock_path.parent().context("retrieve parent from socket path")?).await?;
```

Reply:
done